lib/radiator/directory/editor.ex

```diff
@@ -458,4 +458,89 @@ defmodule Radiator.Directory.Editor do
       @not_authorized_match
     end
   end
+
+  ## User Permission Management
+
+  # TODO: list all collaborators of underlying entities as well.
+
+  @spec list_collaborators(Auth.User.t(), Network.t()) ::
+          [
+            Collaborator.t()
+          ]
+          | {:error, any}
+  def list_collaborators(actor = %Auth.User{}, subject = %Network{}) do
+    if has_permission(actor, subject, :manage) do
+      query =
+        from p in Ecto.assoc(subject, :permissions),
+          join: u in Auth.User,
+          on: p.user_id == u.id,
+          join: s in Network,
+          on: s.id == p.subject_id,
+          preload: [user: u]
+
+      with perm_list when is_list(perm_list) <- Repo.all(query) do
```

Comment:
The guard `is_list(perm_list)` is unnecessary. `Repo.all/1` always returns a list or raises. If you want to check for a nonempty list, I think the recommended way is `with perm_list = [_ | _] <- Repo.all(query)`

Reply:
k